app/src/features/surveys/habitat-features/create/CreateHabitatFeaturePage.tsx

```diff
@@ -0,0 +1,182 @@
+import CircularProgress from '@mui/material/CircularProgress';
+import Container from '@mui/material/Container';
+import Link from '@mui/material/Link';
+import Paper from '@mui/material/Paper';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import PageHeader from 'components/layout/PageHeader';
+import { DialogContext } from 'contexts/dialogContext';
+import { FormikProps } from 'formik';
+import { APIError } from 'hooks/api/useAxios';
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import { useHabitatFeatureTableContext, useProjectContext, useSurveyContext } from 'hooks/useContext';
+import { useUnsavedChangesDialog } from 'hooks/useUnsavedChangesDialog';
+import { useContext, useRef, useState } from 'react';
+import { Prompt, useHistory } from 'react-router';
+import { Link as RouterLink } from 'react-router-dom';
+import {
+  CreateHabitatFeatureFormValues,
+  HabitatFeatureFormContainer
+} from '../components/forms/HabitatFeatureFormContainer';
+
+const initialHabitatFeatureFormValues: CreateHabitatFeatureFormValues = {
+  habitat_feature_type_id: '' as unknown as number,
+  latitude: '' as unknown as number,
+  longitude: '' as unknown as number,
+  count: '' as unknown as number,
+  observed_date: '',
+  observed_time: ''
+};
+
+/**
+ * Page for creating a new habitat feature.
+ *
+ * @return {*} {JSX.Element}
+ */
+export const CreateHabitatFeaturePage = (): JSX.Element => {
+  const biohubApi = useBiohubApi();
+  const history = useHistory();
+  const projectContext = useProjectContext();
+  const surveyContext = useSurveyContext();
+  const habitatFeatureContext = useHabitatFeatureTableContext();
+  const dialogContext = useContext(DialogContext);
+
+  const { locationChangeInterceptor, skipUnsavedChangesDialog } = useUnsavedChangesDialog();
+
+  const formikRef = useRef<FormikProps<CreateHabitatFeatureFormValues>>(null);
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  /**
+   * Handle the create habitat feature form submission.
+   *
+   * @param {CreateHabitatFeatureFormValues} values
+   * @returns {*} {Promise<void>}
+   */
+  const handleSubmit = async (values: CreateHabitatFeatureFormValues): Promise<void> => {
+    try {
+      setIsSubmitting(true);
+      await biohubApi.habitatFeature.createSurveyHabitatFeatures(surveyContext.projectId, surveyContext.surveyId, [
+        {
+          habitat_feature_type_id: values.habitat_feature_type_id,
+          survey_id: surveyContext.surveyId,
+          latitude: values.latitude,
+          longitude: values.longitude,
+          count: values.count,
+          observed_date: values.observed_date,
+          observed_time: values.observed_time
+        }
+      ]);
+
+      skipUnsavedChangesDialog();
+      history.goBack();
+
+      dialogContext.setSnackbar({
+        open: true,
+        snackbarMessage: 'Successfully created habitat feature'
+      });
+
+      await habitatFeatureContext.refreshData();
```

Comment:
Do we need to await this refresh? 

Also, does the history.goBack() need to be balled after? I assume it gets queued up so it works fine as is?

Reply:
Ok I swapped the order of operations for this section, and dropped the await for the refresh.